src/System.Private.CoreLib/shared/System/IO/Path.cs

```diff
@@ -444,6 +461,61 @@ public static string Join(string? path1, string? path2, string? path3)
             return Join(path1.AsSpan(), path2.AsSpan(), path3.AsSpan());
         }
 
+        public static string Join(string? path1, string? path2, string? path3, string? path4)
+        {
+            return Join(path1.AsSpan(), path2.AsSpan(), path3.AsSpan(), path4.AsSpan());
+        }
+
+        public static string Join(params string[] paths)
+        {
+            if (paths == null)
+            {
+                throw new ArgumentNullException(nameof(paths));
+            }
+
+            int finalSize = 0;
+            
+            for (int i = 0; i < paths.Length; i++)
```

Comment:
I realize this code is copied from Combine. However this whole loop just to figure the right size seems wierd, because (1) it doesn't matter if it's a bit too large and (2) the pool is almost certainly going to give you something too large anyway, possibly much too large and (3) if it's too small, and this is getting called a lot, the pool is going to get the larger ones returned to it and will pass them back to the next call [albeit possibly fragmented]

I wonder if we ever measured simply using 200 x entries in the array.

Still, no need to touch it now (and it would need some perf measurement - perhaps that was done and motivated thew way it is)

cc @jeremykuhne

Reply:
@JeremyKuhne  Would it make sense to use valueStringBuilder here with initital buffer as MaxShortPath*arraySize or just MaxShortPath ?

Reply:
We could use `string.Create()` like we do below and avoid any duplicate copies (i.e. just write directly into the string). I don't think we did a good job of measuring, however. Some good perf testing is probably in order.

When you do use `string.Create()` you need to know exactly what size the output string is going to be. When we're using a builder we don't want to undershoot on the size as that will cause more allocations / copies / etc.

I'd be happy to see the `StringBuilderCache` usage go away here. I'd just use `ValueStringBuilder` and set it's capacity large enough after counting the total number of characters (which will rent for you). I'm not sure how much it would win to have a small stack buffer to populate the `ValueStringBuilder`. I wouldn't use anything over `MaxPath` (260) for sure.

If someone wants to explore rewriting to directly write into a string, that would be cool.